sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java

```diff
@@ -500,6 +505,74 @@ public AwsIotMqttConnectionBuilder withWebsocketCredentialsProvider(CredentialsP
         return this;
     }
 
+    /**
+     * A helper function to add parameters to the username in the withCustomAuthorizer function
+     */
+    private String addUsernameParameter(String inputString, String parameterValue, String parameterPreText, Boolean addedQuery) {
+        String return_string = inputString;
+        if (addedQuery == false) {
+            return_string += "?";
+        } else {
+            return_string += "&";
+        }
+
+        if (parameterValue.contains(parameterPreText)) {
+            return return_string + parameterValue;
+        } else {
+            return return_string + parameterPreText + parameterValue;
+        }
+    }
+
+    /**
+     * Configures the MQTT connection so it can use a custom authorizer.
+     * This function will modify the username, port, and TLS options.
+     *
+     * Note: All arguments are optional and can have "null" as valid input.
+     * See the description for each argument for information on what happens if null is passed.
+     * @param username The username to use with the custom authorizer. If null is passed, it will check to
+     *                 see if a username has already been set (via withUsername function). If no username is set then
+     *                 no username will be passed with the MQTT connection.
+     * @param authorizerName The name of the custom authorizer. If null is passed, then 'x-amz-customauthorizer-name'
+     *                       will not be added with the MQTT connection.
+     * @param signature The signature of the custom authorizer. If null is passed, then 'x-amz-customauthorizer-signature'
+     *                  will not be added with the MQTT connection.
+     * @param password The password to use with the custom authorizer. If null is passed, then no password will be set.
+     * @return
+     */
+    public AwsIotMqttConnectionBuilder withCustomAuthorizer(String username, String authorizerName, String signature, String password) {
+        isUsingCustomAuthorizer = true;
+        String username_string = "";
+        Boolean username_added_query = false;
+
+        if (username == null) {
+            if (config.getUsername() != null) {
+                username_string += config.getUsername();
+            }
+        } else {
+            username_string += username;
+        }
+
+        if (authorizerName != null) {
+            username_string = addUsernameParameter(username_string, authorizerName, "x-amz-customauthorizer-name=", username_added_query);
+            username_added_query = true;
+        }
+        if (signature != null) {
+            username_string = addUsernameParameter(username_string, signature, "x-amz-customauthorizer-signature=", username_added_query);
+            username_added_query = true;
```

Comment:
username_added_query doesn't appear to be used beyond this point so we may be able to remove this true assignment.